woocommerce/woocommerce-ios
# [Custom Amounts M1] Tracking

Closes: #10881


## Description

With this PR we implement tracking for the first milestone of Custom Amounts.

## Testing instructions


- Go to Orders.
- Tap on + to create a new order.
- Tap on Add custom amount. The event should be tracked:
`🔵 Tracked order_creation_add_custom_amount_tapped, properties: [AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com", AnyHashable("is_wpcom_store"): true, AnyHashable("was_ecommerce_trial"): false, AnyHashable("blog_id"): 214354650, AnyHashable("plan"): "business-bundle"]`
- Add an amount and tap on Add custom amount.  The done button tapped event should be tracked:
` 🔵 Tracked add_custom_amount_done_tapped, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com", AnyHashable("is_wpcom_store"): true, AnyHashable("was_ecommerce_trial"): false, AnyHashable("plan"): "business-bundle"]`
- If you added a name, also the name event should be tracked:
`🔵 Tracked add_custom_amount_name_added, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("plan"): "business-bundle", AnyHashable("blog_id"): 214354650, AnyHashable("was_ecommerce_trial"): false, AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com"]`
After the custom amount is added to the order as a fee, the event is tracked:
`🔵 Tracked order_fee_add, properties: [AnyHashable("plan"): "business-bundle", AnyHashable("flow"): "creation", AnyHashable("blog_id"): 214354650, AnyHashable("is_wpcom_store"): true, AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com", AnyHashable("was_ecommerce_trial"): false]`
- Tap to edit the custom amount. The event should be tracked:
`🔵 Tracked order_creation_edit_custom_amount_tapped, properties: [AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com", AnyHashable("plan"): "business-bundle", AnyHashable("was_ecommerce_trial"): false, AnyHashable("blog_id"): 214354650, AnyHashable("is_wpcom_store"): true]`
- After you edit the custom amount and tap on Add custom amount to submit the update, the event is tracked:
`🔵 Tracked order_fee_update, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("is_wpcom_store"): true, AnyHashable("plan"): "business-bundle", AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com", AnyHashable("was_ecommerce_trial"): false, AnyHashable("flow"): "creation"]`
- Tap to remove the custom amount. The relevant should be tracked:
`🔵 Tracked order_creation_remove_custom_amount_tapped, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("was_ecommerce_trial"): false, AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com", AnyHashable("blog_id"): 214354650, AnyHashable("plan"): "business-bundle"]`
`🔵 Tracked order_fee_remove, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): 214354650, AnyHashable("flow"): "creation", AnyHashable("plan"): "business-bundle", AnyHashable("was_ecommerce_trial"): false, AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com"]`



## Screenshots

N/A


---
